### PR TITLE
[12.0][ADD] sale_expense_receipt

### PR DESCRIPTION
**Pull request by Khalid-SerpentCS**
Sale Expense Receipt
===================================================================================================
This module allows you to attach all the receipts of the expenses when invoicing a sale order and send them with the invoice.

Usage
=================================================================================================
* Go to Expenses
* Create an expense with a product that is re-invoiced (Re-invoicing policy: At cost)
* Select a sale order and attach the receipt
* Submit, approve and post the journal entries
* Go to Sales > Sales Orders
* Open the sale orders selected on the expense and create the invoice. A file "Expense_Receipt.zip" should be attached to the 
  invoice and should contain the receipt of the expense.

### CODE REVIEW

File: sale_expense_receipt/models/sale.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                lambda e: e.sheet_id.state == 'post' and
+                e.product_id == line.product_id and
+                e.unit_amount == line.price_unit)
+
+        attachments = self.env['ir.attachment'].search([
+            ('res_id', 'in', expenses.ids)])
+
+        if attachments:
+            file_path = pathlib.Path(temp_dir + '/Expense_Receipt.zip')
+            receipt_zip_file = zipfile.ZipFile(
```
Comment (len-foss) [inline sale_expense_receipt/models/sale.py:35]: Why not use a `with` here?

File: sale_expense_receipt/models/sale.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                file_path, mode='w')
+            for attachment in attachments:
+                receipt_zip_file.writestr(attachment.name,
+                                          base64.b64decode(attachment.datas))
```
Comment (len-foss) [inline sale_expense_receipt/models/sale.py:39]: Maybe `from base64 import b64decode` to lighten a bit the lines?

**Review by len-foss — APPROVED** [review]
Code review: LGTM
However, you would need to add some tests to pass the required code coverage.

**Comment by rousseldenis** [comment]
@JayVora-SerpentCS

**Comment by JayVora-SerpentCS** [comment]
> @JayVora-SerpentCS

do we need this feature? It was long back migrated.

**Pull request closed without merging**